src/Concordium/Client/GRPC2.hs

```diff
@@ -3106,21 +3116,21 @@ withGRPCCoreV2 helper k = do
                     let response = toGRPCResult ret'
                     addHeaders response
                     return $ k response
-                else return $ k (Left "Cannot establish connection to GRPC endpoint.")
+                else return $ k (RequestFailed "Cannot establish connection to GRPC endpoint.")
         (_, Just v) ->
             let response = toGRPCResult' v
              in do
                     addHeaders response
                     return (k response)
   where
     addHeaders response = case response of
-        Right GRPCResponse{..} -> do
+        StatusOk GRPCResponse{..} -> do
```

Comment:
Should this not be done for non-'OK' error messages as well?

Reply:
If we can, yes.

Reply:
Adding a TODO on Zenhub for this.